### PR TITLE
Allow new import syntax

### PR DESCRIPTION
**Pull request by odersky**
- Replace wildcard import `_` with `*`, which is what basically all other languages use
 - Replace renaming `=>` with a soft keyword `as`.
 - Allow `as` outside braces, as in
     ```scala
     import scala.collection.mutable as mut
     import NumPy as np
     ```

From the doc page:

The syntax of wildcard and renaming imports (and exports) has changed.

## Wildcard Imports

Wildcard imports are now expressed with `*` instead of underscore. Example:
```scala
import scala.annotation.*  // imports everything in the annotation package
```

If you want to import a member named `*` specifically, you can use backticks around it.

```scala
object A:
   def * = ...
   def min = ...

object B:
  import A.`*`   // imports just `*`

object C:
  import A.*     // imports everything in A
```

## Renaming Imports

To rename or exclude an import, we now use `as` instead of `=>`. A single renaming import no longer needs to be enclosed in braces. Examples:

```scala
import A.{min as minimum, `*` as multiply}
import Predef.{augmentString as _, *}     // imports everything except augmentString
import scala.annotation as ann
import java as j
```

### Migration

To support cross-building, Scala 3.0 supports the old import syntax with `_` for wildcards and `=>` for renamings in addition to the new one. The old syntax
will be dropped in a future versions. Automatic rewritings from old to new syntax
are offered under settings `-source 3.1-migration -rewrite`.

### Syntax

```
Import            ::=  ‘import’ ImportExpr {‘,’ ImportExpr}
Export            ::=  ‘export’ ImportExpr {‘,’ ImportExpr}
ImportExpr        ::=  SimpleRef {‘.’ id} ‘.’ ImportSpec
ImportSpec        ::=  NamedSelector
                    |  WildcardSelector
                    | ‘{’ ImportSelectors) ‘}’
NamedSelector     ::=  id [‘as’ (id | ‘_’)]
WildCardSelector  ::=  ‘*' | ‘given’ [InfixType]
ImportSelectors   ::=  NamedSelector [‘,’ ImportSelectors]
                    |  WildCardSelector {‘,’ WildCardSelector}
```

### CODE REVIEW

**Comment by odersky** [comment]
Based on #11240

**Comment by odersky** [comment]
Exports still have to be adapted to that we can do simple identifier renames such as
```scala
export foo as bar
```

**Comment by odersky** [comment]
@liufengyun Can you review the technical side here? It looks like the proposal has string support, so we will probably go ahead and merge this.

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 1]
```diff
@@ -3097,48 +3097,62 @@ object Parsers {
...
-    def importExpr(mkTree: ImportConstr): () => Tree = {
-
-      /** '_' */
-      def wildcardSelectorId() = atSpan(in.skipToken()) { Ident(nme.WILDCARD) }
-      def givenSelectorId(start: Offset) = atSpan(start) { Ident(nme.EMPTY) }
+    /** ImportExpr       ::=  SimpleRef {‘.’ id} ‘.’ ImportSpec
+     *                     |  SimpleRef ‘as’ id
+     *  ImportSpec       ::=  NamedSelector
+     *                     |  WildcardSelector
+     *                     | ‘{’ ImportSelectors) ‘}’
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala:3104]:
```suggestion
     *                     | ‘{’ ImportSelectors ‘}’
```

File: compiler/src/dotty/tools/dotc/parsing/Parsers.scala [hunk 2]
```diff
@@ -3097,48 +3097,62 @@ object Parsers {
...
+          report.errorOrMigrationWarning(
+            em"`_` is no longer supported for a wildcard import; use `*` instead${rewriteNotice("3.1")}",
+            in.sourcePos())
+          patch(source, Span(in.offset, in.offset + 1), "*")
+        ImportSelector(atSpan(in.skipToken()) { Ident(nme.WILDCARD) })
+
+      /** 'given [InfixType]' */
+      def givenSelector() =
+        ImportSelector(
+          atSpan(in.skipToken()) { Ident(nme.EMPTY )},
```
Comment (liufengyun) [inline compiler/src/dotty/tools/dotc/parsing/Parsers.scala:3124]:
```suggestion
          atSpan(in.skipToken()) { Ident(nme.EMPTY) },
```

**Review by liufengyun — APPROVED** [review]
LGTM

**Pull request merged**